File: 401. Binary Watch.cpp
class Solution {
public:
    vector<string> readBinaryWatch(int turnedOn) {
        vector<string>ans;
        for(int hr=0;hr<=11;hr++){
            for(int min=0;min<=59;min++){
                if(__builtin_popcount(hr) + __builtin_popcount(min) == turnedOn){
                    string res = "";
                    if(min==0 || min==1 || min==2 || min==3 || min==4 || min==5 || min==6 || min==7 || min==8 || min==9){
                        res = to_string(hr) + ":" + "0" + to_string(min);
                    }
                    else{
                        res = to_string(hr) + ":" + to_string(min);
                    }
                    ans.push_back(res);
                }
            }
        }
        return ans;
    }
};

File: 401. Binary Watch.ts
function countBits(n: number): number {
    let cnt: number = 0;
    while(n > 0){
        n = n & (n - 1);
        cnt++;
    }
    return cnt;
}
function readBinaryWatch(turnedOn: number): string[] {
     let ans: string[] = [];

    for(let hr = 0; hr <= 11; hr++){
        for(let min = 0; min <= 59; min++){

            if(countBits(hr) + countBits(min) === turnedOn){

                let res: string = "";

                if(min < 10){
                    res = hr + ":" + "0" + min;
                }
                else{
                    res = hr + ":" + min;
                }

                ans.push(res);
            }
        }
    }

    return ans;
};